hibernate-core/src/test/java/org/hibernate/orm/test/jpa/query/TupleQueryTest.java

```diff
@@ -85,6 +91,28 @@ public void testGetAlias(EntityManagerFactoryScope scope) {
 		);
 	}
 
+	@Test
+	@JiraKey( "HHH-16742" )
+	public void testTwoDifferentAliasesToSameColumn(EntityManagerFactoryScope scope) {
+		scope.inTransaction(
+				entityManager -> {
+					TypedQuery<Tuple> query = entityManager.createQuery( "SELECT u.firstName as fn, u.firstName as fn_2 from User u", Tuple.class );
+
+					Tuple result = query.getResultList().get(0);
+					assertNotNull( result );
+					assertEquals( 2, result.getElements().size() );
+					try {
+						// Should not throw IllegalArgumentException
+						result.get("fn", String.class);
+						result.get("fn_2", String.class);
+					}
+					catch (IllegalArgumentException iae) {
```

Comment:
Please remove the try-catch part. If an exception is thrown, the test will fail anyway.